tests/e2e-evm/contracts/ABI_BasicTests.sol

```diff
@@ -18,12 +18,67 @@ contract Caller {
 
             // solhint-disable-next-line no-inline-assembly
             assembly {
+                // Bubble up errors: revert(pointer, length of revert reason)
+                // - result is a dynamic array, so the first 32 bytes is the
+                //   length of the array.
+                // - add(32, result) skips the length of the array and points to
+                //   the start of the data.
+                // - mload(result) reads 32 bytes from the memory location,
+                //   which is the length of the revert reason.
                 revert(add(32, result), mload(result))
             }
         }
     }
 
-    // TODO: Callcode
+    function functionCallCode(address to, bytes calldata data) external payable {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            // Copy the calldata to memory, as callcode uses memory pointers.
```

Comment:
💯 and good note on memory safety for future reference if modifying this function